src/platform/packages/shared/kbn-unified-metrics-grid/src/components/metrics_grid.tsx

```diff
@@ -8,143 +8,82 @@
 
   const chartSize = useMemo(() => (columns === 2 || columns === 4 ? 's' : 'm'), [columns]);
 
-  if (loading) {
-    return (
-      <EuiFlexGroup
-        justifyContent="center"
-        alignItems="center"
-        css={css`
-          min-height: 400px;
-        `}
-      >
-        <EuiFlexItem grow={false}>
-          <EuiLoadingChart size="m" />
-        </EuiFlexItem>
-      </EuiFlexGroup>
-    );
-  }
+  const colorPalette = useMemo(
+    () => Object.values(euiTheme.colors.vis).slice(0, 10),
+    [euiTheme.colors.vis]
+  );
 
-  if (pivotOn === 'metric' && fields.length === 0) {
-    return (
-      <div
-        css={css`
-          width: 100%;
-          height: 100%;
-        `}
-      >
-        <EuiFlexGroup
-          direction="column"
-          alignItems="center"
-          justifyContent="spaceAround"
-          css={css`
-            height: 100%;
-          `}
-          gutterSize="s"
-        >
-          <EuiFlexItem
-            css={css`
-              justify-content: end;
-            `}
-          >
-            <EuiIcon type={IconChartBarStacked as IconType} size="l" />
-          </EuiFlexItem>
-          <EuiFlexItem>
-            <EuiText size="xs">
-              {i18n.translate('metricsExperience.grid.noData', {
-                defaultMessage: 'No results found',
-              })}
-            </EuiText>
-          </EuiFlexItem>
-        </EuiFlexGroup>
-      </div>
-    );
-  }
+  const rows = useMemo(() => {
+    return pivotOn === 'metric'
+      ? fields.map((field, i) => ({ key: `${field.name}-${i}`, metric: field }))
+      : dimensions.map((dim, i) => ({ key: `${dim}-${i}`, metric: fields }));
+  }, [pivotOn, fields, dimensions]);
 
   return (
-    <EuiFlexGrid
-      columns={columns}
-      gutterSize="s"
-      css={css`
-        overflow: auto;
-        padding: ${euiTheme.size.s} ${euiTheme.size.s} 0;
-        ${euiScrollBarStyles(euiThemeContext)}
-      `}
-      data-test-subj="unifiedMetricsExperienceGrid"
-    >
-      {pivotOn === 'metric'
-        ? fields.map((field, index) => (
-            <EuiFlexItem key={field.name}>
-              <MetricChart
-                metric={field}
-                timeRange={timeRange}
-                dimensions={dimensions}
-                filters={filters}
-                colorIndex={index}
-                size={chartSize}
-              />
-            </EuiFlexItem>
-          ))
-        : dimensions.map((dimension, index) => (
-            <EuiFlexItem key={dimension}>
-              <MetricChart
-                metric={fields}
-                timeRange={timeRange}
-                byDimension={dimension}
-                dimensions={[dimension]}
-                filters={filters}
-                colorIndex={index}
-                size={chartSize}
-              />
-            </EuiFlexItem>
-          ))}
+    <EuiFlexGrid columns={columns} gutterSize="s" data-test-subj="unifiedMetricsExperienceGrid">
```

Comment:
Nit: Do we still need `unified` in the name? We kept `metricsExperience` everywhere else.